src/utils/perfherder/chartJs/perfherderFormatter.js

```diff
@@ -7,33 +7,78 @@ const dataToChartJSformat = data => data.map(({ datetime, value }) => ({
   y: value,
 }));
 
-const generateInitialOptions = (meta) => {
-  const higherIsBetter = (meta.lower_is_better === false);
+const fetchPushidData = async (url) => {
+  try {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+  } catch (err) {
+    return err;
+  }
+};
+
+const generateInitialOptions = (series) => {
+  const higherIsBetter = !series.meta.lower_is_better;
+  const higherOrLower = higherIsBetter ? 'higher is better' : 'lower is better';
   return {
     reverse: higherIsBetter,
     scaleLabel: higherIsBetter ? 'Score' : 'Load time',
     tooltips: {
       callbacks: {
         footer: (tooltipItems, data) => {
+          const tooltipData = [];
           let delta = 'n/a';
+          let deltaPercentage = 'n/a';
+          let jobLink = '';
+          let hgLink = '';
+          let dataset = 'n/a';
+          let currentData = 'n/a';
           if (tooltipItems[0].index > 0) {
-            delta = (data.datasets[tooltipItems[0].datasetIndex].data[tooltipItems[0].index].y
-              - data.datasets[tooltipItems[0].datasetIndex].data[tooltipItems[0].index - 1].y)
-              .toFixed(2);
-            }
-            return `Delta: ${delta}`;
-          },
+            dataset = data.datasets[tooltipItems[0].datasetIndex].data;
+
+            currentData = dataset[tooltipItems[0].index].y;
+            const previousData = dataset[tooltipItems[0].index - 1].y;
+
+            delta = (currentData - previousData).toFixed(2);
+            deltaPercentage = (((currentData - previousData) / previousData) * 100).toFixed(2);
+
+            const tooltipDataSet = series.data[tooltipItems[0].index];
+            // make job link against job id
+            jobLink = `https://treeherder.mozilla.org/#/jobs?repo=mozilla-central&revision=ca0f00593e38cdab54c3a990059bbf1150e77365&selectedJob=${tooltipDataSet.job_id}&group_state=expanded`;
+
+            // generate hg link
+            const pushId = tooltipDataSet.push_id;
+            const previousPushID = series.data[tooltipItems[0].index - 1].push_id;
+
+            const pushiIdLink = `https://treeherder.mozilla.org/api/project/mozilla-central/resultset/${pushId}/?format=json`;
+            const previousPushiIdLink = `https://treeherder.mozilla.org/api/project/mozilla-central/resultset/${previousPushID}/?format=json`;
+
+            // get push data against pushid
+            const pushIdData = fetchPushidData(pushiIdLink);
+            const previousPushIdData = fetchPushidData(previousPushiIdLink);
```

Comment:
I think you want to do here is wrap both of these calls in `await Promise.all` so that both are executed in parallel but the code after line 58 isn't executed until those responses have returned.

Reply:
Hi @sarah-clements, thanks for suggesting, it works better but the actual problem is still there. 

>> It returns the data as expected and i can see that doing console.log()
but it doesn't appear on tooltip and makes it `[object promise]`
<img width="173" alt="screen shot 2019-01-28 at 5 45 38 pm" src="https://user-images.githubusercontent.com/13648547/51841549-be52a380-2330-11e9-9bb2-3022db7ce129.png">

I tried to find the solution for this over internet but it looks like i am missing some concept here. 
I'm unable to find any good reason for this. Any thoughts?